diff -ur comm-1.9.1/mozilla/modules/libpref/src/init/all.js comm-1.9.1.new/mozilla/modules/libpref/src/init/all.js
--- comm-1.9.1/mozilla/modules/libpref/src/init/all.js	2009-12-05 02:56:20.000000000 +0100
+++ comm-1.9.1.new/mozilla/modules/libpref/src/init/all.js	2009-12-21 09:16:41.372169512 +0100
@@ -1240,9 +1240,9 @@
 pref("font.name.monospace.th", "Tahoma");
 pref("font.name.cursive.th", "Tahoma");
 
-pref("font.name.serif.tr", "Times New Roman");
-pref("font.name.sans-serif.tr", "Arial");
-pref("font.name.monospace.tr", "Courier New");
+pref("font.name.serif.tr", "serif");
+pref("font.name.sans-serif.tr", "sans-serif");
+pref("font.name.monospace.tr", "monospace");
 pref("font.name.cursive.tr", "Comic Sans MS");
 
 pref("font.name.serif.x-baltic", "Times New Roman");
@@ -1250,9 +1250,9 @@
 pref("font.name.monospace.x-baltic", "Courier New");
 pref("font.name.cursive.x-baltic", "Comic Sans MS");
 
-pref("font.name.serif.x-central-euro", "Times New Roman");
-pref("font.name.sans-serif.x-central-euro", "Arial");
-pref("font.name.monospace.x-central-euro", "Courier New");
+pref("font.name.serif.x-central-euro", "serif");
+pref("font.name.sans-serif.x-central-euro", "sans-serif");
+pref("font.name.monospace.x-central-euro", "monospace");
 pref("font.name.cursive.x-central-euro", "Comic Sans MS");
 
 pref("font.name.serif.x-cyrillic", "Times New Roman");
@@ -1260,9 +1260,9 @@
 pref("font.name.monospace.x-cyrillic", "Courier New");
 pref("font.name.cursive.x-cyrillic", "Comic Sans MS");
 
-pref("font.name.serif.x-unicode", "Times New Roman");
-pref("font.name.sans-serif.x-unicode", "Arial");
-pref("font.name.monospace.x-unicode", "Courier New");
+pref("font.name.serif.x-unicode", "serif");
+pref("font.name.sans-serif.x-unicode", "sans-serif");
+pref("font.name.monospace.x-unicode", "monospace");
 pref("font.name.cursive.x-unicode", "Comic Sans MS");
 
 pref("font.name.serif.x-western", "Times New Roman");
@@ -1396,23 +1396,23 @@
 
 pref("font.default.ar", "sans-serif");
 pref("font.size.variable.ar", 16);
-pref("font.size.fixed.ar", 13);
+pref("font.size.fixed.ar", 14);
 
-pref("font.default.el", "serif");
+pref("font.default.el", "sans-serif");
 pref("font.size.variable.el", 16);
-pref("font.size.fixed.el", 13);
+pref("font.size.fixed.el", 14);
 
 pref("font.default.he", "sans-serif");
 pref("font.size.variable.he", 16);
-pref("font.size.fixed.he", 13);
+pref("font.size.fixed.he", 14);
 
 pref("font.default.ja", "sans-serif");
 pref("font.size.variable.ja", 16);
-pref("font.size.fixed.ja", 16);
+pref("font.size.fixed.ja", 14);
 
 pref("font.default.ko", "sans-serif");
 pref("font.size.variable.ko", 16);
-pref("font.size.fixed.ko", 16);
+pref("font.size.fixed.ko", 14);
 
 pref("font.default.th", "serif");
 pref("font.size.variable.th", 16);
@@ -1856,22 +1856,22 @@
 pref("font.size.variable.ko", 16);
 pref("font.size.fixed.ko", 16);
 
-pref("font.default.th", "serif");
+pref("font.default.th", "sans-serif");
 pref("font.size.variable.th", 16);
-pref("font.size.fixed.th", 13);
+pref("font.size.fixed.th", 14);
 pref("font.minimum-size.th", 10);
 
-pref("font.default.tr", "serif");
+pref("font.default.tr", "sans-serif");
 pref("font.size.variable.tr", 16);
-pref("font.size.fixed.tr", 13);
+pref("font.size.fixed.tr", 14);
 
 pref("font.default.x-armn", "serif");
 pref("font.size.variable.x-armn", 16);
 pref("font.size.fixed.x-armn", 13);
 
-pref("font.default.x-baltic", "serif");
+pref("font.default.x-baltic", "sans-serif");
 pref("font.size.variable.x-baltic", 16);
-pref("font.size.fixed.x-baltic", 13);
+pref("font.size.fixed.x-baltic", 14);
 
 pref("font.default.x-beng", "serif");
 pref("font.size.variable.x-beng", 16);
@@ -1881,9 +1881,9 @@
 pref("font.size.variable.x-cans", 16);
 pref("font.size.fixed.x-cans", 13);
 
-pref("font.default.x-central-euro", "serif");
+pref("font.default.x-central-euro", "sans-serif");
 pref("font.size.variable.x-central-euro", 16);
-pref("font.size.fixed.x-central-euro", 13);
+pref("font.size.fixed.x-central-euro", 14);
 
 pref("font.default.x-cyrillic", "serif");
 pref("font.size.variable.x-cyrillic", 16);
@@ -2126,9 +2126,9 @@
 pref("font.size.variable.x-central-euro", 16);
 pref("font.size.fixed.x-central-euro", 13);
 
-pref("font.default.x-cyrillic", "serif");
+pref("font.default.x-cyrillic", "sans-serif");
 pref("font.size.variable.x-cyrillic", 16);
-pref("font.size.fixed.x-cyrillic", 13);
+pref("font.size.fixed.x-cyrillic", 14);
 
 pref("font.default.x-devanagari", "serif");
 pref("font.size.variable.x-devanagari", 16);
@@ -2138,25 +2138,25 @@
 pref("font.size.variable.x-tamil", 16);
 pref("font.size.fixed.x-tamil", 13);
 
-pref("font.default.x-unicode", "serif");
+pref("font.default.x-unicode", "sans-serif");
 pref("font.size.variable.x-unicode", 16);
-pref("font.size.fixed.x-unicode", 13);
+pref("font.size.fixed.x-unicode", 14);
 
-pref("font.default.x-western", "serif");
+pref("font.default.x-western", "sans-serif");
 pref("font.size.variable.x-western", 16);
-pref("font.size.fixed.x-western", 13);
+pref("font.size.fixed.x-western", 14);
 
-pref("font.default.zh-CN", "serif");
+pref("font.default.zh-CN", "sans-serif");
 pref("font.size.variable.zh-CN", 16);
-pref("font.size.fixed.zh-CN", 16);
+pref("font.size.fixed.zh-CN", 14);
 
-pref("font.default.zh-TW", "serif");
+pref("font.default.zh-TW", "sans-serif");
 pref("font.size.variable.zh-TW", 16);
-pref("font.size.fixed.zh-TW", 16);
+pref("font.size.fixed.zh-TW", 14);
 
-pref("font.default.zh-HK", "serif");
+pref("font.default.zh-HK", "sans-serif");
 pref("font.size.variable.zh-HK", 16);
-pref("font.size.fixed.zh-HK", 16);
+pref("font.size.fixed.zh-HK", 14);
 
 pref("netinst.profile.show_profile_wizard", true); 
 
