--- icedove-3.1.2/mozilla/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc~	2010-08-03 02:28:30.000000000 +0300
+++ icedove-3.1.2/mozilla/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc	2010-08-17 18:39:51.092556800 +0300
@@ -43,6 +43,7 @@
 #include <string.h>
 #include <sys/mman.h>
 #include <unistd.h>
+#include <sys/stat.h>
 
 #include <algorithm>
 #include <cassert>
