--- icedove-3.1.2/mozilla/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc~	2010-08-03 02:28:30.000000000 +0300
+++ icedove-3.1.2/mozilla/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc	2010-08-17 18:39:51.092556800 +0300
@@ -43,6 +43,7 @@
 #include <string.h>
 #include <sys/mman.h>
 #include <unistd.h>
+#include <sys/stat.h>
 
 #include <algorithm>
 #include <cassert>
--- icedove-3.1.2/mozilla/mozilla/toolkit/crashreporter/google-breakpad/src/tools/linux/dump_syms/Makefile.in~	2010-08-17 18:24:25.755776132 +0300
+++ icedove-3.1.2/mozilla/mozilla/toolkit/crashreporter/google-breakpad/src/tools/linux/dump_syms/Makefile.in	2010-08-17 19:16:17.356296784 +0300
@@ -62,8 +62,4 @@
 CPP_PROG_LINK = 1
 FORCE_USE_PIC = 1
 
-#XXX: bug 554854 causes us to be unable to run binaries on the build slaves
-# due to them having an older libstdc++
-HOST_LDFLAGS += -static
-
 include $(topsrcdir)/config/rules.mk
