--- mozilla/editor/txtsvc/src/nsTextServicesDocument.cpp~	2006-01-19 10:35:20.000000000 +0000
+++ mozilla/editor/txtsvc/src/nsTextServicesDocument.cpp	2006-01-19 10:38:19.000000000 +0000
@@ -2501,6 +2501,8 @@
 nsresult
 nsTextServicesDocument::DeleteNode(nsIDOMNode *aChild)
 {
+  NS_ENSURE_TRUE(mIterator, NS_ERROR_FAILURE);
+
   //**** KDEBUG ****
   // printf("** DeleteNode: 0x%.8x\n", aChild);
   // fflush(stdout);
